CIME/XML/archive_base.py

```diff
@@ -139,21 +139,28 @@ def get_all_hist_files(self, casename, model, from_dir, suffix="", ref_case=None
         for ext in extensions:
             if ext.endswith("$") and has_suffix:
                 ext = ext[:-1]
-            string = model + r"\d?_?(\d{4})?(_d\d{2})?\." + ext
+
+            # Build the regex that will identify history files
+            re_string = model  # We expect the model name to be in the filename
+            re_string += (
+                r"\d?_?(\d{4})?(_d\d{2})?"  # Need this for potential DART files?
```

Comment:
It's not entirely clear to me why this weird digit stuff has to be injected, but this has been in there in one form or another for a very long time.